libs/test-common/src/main/java/org/codice/ddf/test/common/mockito/PrivilegedVerificationMode.java

```diff
@@ -0,0 +1,142 @@
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged(VerificationMode mode) {
+    return new PrivilegedVerificationMode(mode);
+  }
+
+  /**
+   * Returns a {@link VerificationMode} object that ensures a single expected calls was made inside
+   * a {@code AccessController.doPrivileged()} block.
+   *
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged() {
+    return new PrivilegedVerificationMode(times(1));
+  }
+
+  @Override
+  public void verify(VerificationData verificationData) {
+    String target = verificationData.getTarget().toString();
+
+    mode.verify(verificationData);
+
+    verify(verificationData.getTarget(), verificationData.getAllInvocations());
+  }
+
+  @Override
+  public void verifyInOrder(VerificationDataInOrder verificationDataInOrder) {
+    String target = verificationDataInOrder.getWanted().toString();
+
+    if (!(mode instanceof VerificationInOrderMode)) {
+      throw new MockitoAssertionError(mode.toString() + " cannot be used in ordered mode");
+    }
+
+    ((VerificationInOrderMode) mode).verifyInOrder(verificationDataInOrder);
+
+    verify(verificationDataInOrder.getWanted(), verificationDataInOrder.getAllInvocations());
+  }
+
+  @Override
+  public VerificationMode description(String description) {
+    return VerificationModeFactory.description(this, description);
+  }
+
+  private void verify(MatchableInvocation wanted, List<Invocation> invocations) {
+
+    if (invocations.size() == 0) {
+      return;
+    }
+
+    try {
+      Location location = invocations.get(0).getLocation();
+      Field stackTraceHolderField = getStackTraceHolderField(location);
+
+      for (Invocation invocation : invocations) {
+        if (wanted.matches(invocation)) {
+          if (getStackTraceElements(stackTraceHolderField, invocation)
+              .noneMatch(this::isDoPrivilegedCall)) {
+            throw new MockitoAssertionError(
+                wanted.toString() + " not called in a doPrivileged block");
+          }
+        }
+      }
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      throw new MockitoAssertionError(
+          "Failed to verify that "
+              + wanted.toString()
+              + " was called in a doPrivileged block, most likely because Mockito's"
+              + " VerificationData implementation has changed.");
+    }
+  }
+
+  private Stream<StackTraceElement> getStackTraceElements(
+      Field stackTraceHolderField, Invocation invocation) throws IllegalAccessException {
+    return Arrays.stream(
+        ((Throwable) stackTraceHolderField.get(invocation.getLocation())).getStackTrace());
+  }
+
+  private Field getStackTraceHolderField(Location location) throws NoSuchFieldException {
+    Field stackTraceHolderField = location.getClass().getDeclaredField("stackTraceHolder");
```

Comment:
❗️ Since you are in fact relying on an internal implementation class of Mockito, why not actually testing if it is an instance of `org.mockito.internal.debugging.LocationImpl`? In addition, it would kind of explain where the `stackTraceHolder` field name came from.

Might also be a good candidate for contributing or requesting a `getStackTrace()` method to Mockito's `Location` interface.

Reply:
Added a check and posted on Mockito's Google group.